### PR TITLE
Add testing and CI

### LINKED ISSUE #2: Add tests

### LINKED ISSUE #3: Add declarative labels

### PR DESCRIPTION
**Pull request by devpow112**
This adds some of the standard stuff (CI, linting, formatting, license checking, etc) to the repository.

Recommend viewing with white-space diff turned off https://github.com/Brightspace/bunyan-firehose/pull/7/files?diff=unified&w=1. All changes to js files were from auto formatting.

Resolves #2
Resolves #3

### CODE REVIEW

File: .nycrc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+	"all": true,
+	"check-coverage": true,
+	"lines": 0,
```
Comment (devpow112) [inline .nycrc:4]: 0 for now since there are no real tests

File: src/firehoseStream.js [hunk 2]
```diff
@@ -1,156 +1,157 @@
...
+
+		if (this.timer) {
+			clearTimeout(this.timer);
+		}
+
+		if (this.recordsQueue.length >= this.buffer.length || hasPriority) {
+			this.flush();
+
+		} else {
+			this.timer = setTimeout(this.flush.bind(this), this.buffer.timeout * 1000);
```
Comment (ryoung-d2l) [inline src/firehoseStream.js:114]: super nitpicky but a lot of extra whitespace

Comment (devpow112) [reply src/firehoseStream.js:114]: True. This is part of the original source. I was trying to not modify it too much right now. I will clean that up in a future PR as I do more work on changing the source.

Comment (ryoung-d2l) [reply src/firehoseStream.js:114]: No worries, sounds good

**Pull request merged**